deploy-agent/deployd/agent.py

```diff
@@ -51,6 +53,51 @@ class AgentRunMode(object):
     def is_serverless(mode):
         return AgentRunMode.SERVERLESS == mode
 
+class TimeElapsed():
+    def __init__(self):
+        """ keep track of elapsed time in seconds """
+        self._time_start = self._timer()
+        self._time_now = None
+        self._time_elapsed = float(0)
+        self._time_pause = None
+
+    def __call__(self):
+        """ total elapsed running time in seconds """
+        if self._paused():
+            return self._time_elapsed
+        self._time_now = self._timer()
+        self._time_elapsed += float(self._time_now - self._time_start)
+        self._time_start = self._time_now
+        return self._time_elapsed
+
+    def _paused(self):
+        """ timer pause state """
+        if self._time_pause:
+            return True
+        return False
+
+    @staticmethod
+    def _timer():
```

Comment:
given the precision here is in seconds, I think we actually just one time.time()